test/object-store/sync/app.cpp

```diff
@@ -344,10 +304,10 @@ TEST_CASE("app: login_with_credentials integration", "[sync][app][user][baas]")
         int subscribe_processed = 0;
         auto token = app->subscribe([&subscribe_processed](auto& app) {
             if (!subscribe_processed) {
-                REQUIRE(app.current_user());
+                REQUIRE(app.backing_store()->get_current_user());
```

Comment:
App has `current_user()` still. I think all of the places which were doing `app->sync_manager()->all_users()` should also just be `app->all_users()`.

It's probably not immediately possible without more work than it's worth, but ideally the backing store is an implementation detail rather than something the SDKs are using directly, so that the backing store API can be focused on being something that's easy to implement rather than also having to worry about being easy to use. To fit with that, tests which aren't specifically testing the backing store should ideally not have to mention it either.